aptos-move/aptos-aggregator/src/resolver.rs

```diff
@@ -18,29 +28,89 @@ pub enum AggregatorReadMode {
 /// Returns a value of an aggregator from cache or global storage.
 ///   - Ok(..)       if aggregator value exists
 ///   - Err(..)      otherwise.
-pub trait AggregatorResolver {
+pub trait TAggregatorView {
+    type Identifier;
+
+    fn get_aggregator_v1_state_value(
+        &self,
+        id: &Self::Identifier,
+    ) -> anyhow::Result<Option<StateValue>>;
+
+    fn get_aggregator_v1_value(&self, id: &Self::Identifier) -> anyhow::Result<u128> {
+        let maybe_state_value = self.get_aggregator_v1_state_value(id)?;
+        // TODO: consider reviving Option<u128>?
+        let u128_bytes = maybe_state_value
+            .expect("Aggregator V1 cannot be deleted")
+            .into_bytes();
+        bcs::from_bytes(&u128_bytes)
+            .map_err(|_| anyhow::Error::msg("Failed to deserialize aggregator value to u128"))
+    }
+
+    // TODO: Nested options are used due to cyclic dependency. Aggregator crate needs refactoring.
+    fn get_aggregator_v1_state_value_metadata(
+        &self,
+        id: &Self::Identifier,
+    ) -> anyhow::Result<Option<Option<StateValueMetadata>>> {
+        let maybe_state_value = self.get_aggregator_v1_state_value(id)?;
+        Ok(maybe_state_value.map(StateValue::into_metadata))
+    }
+
     /// Returns a value of an aggregator.
-    fn resolve_aggregator_value(
+    fn get_aggregator_v2_value(
         &self,
-        id: &AggregatorID,
-        mode: AggregatorReadMode,
-    ) -> Result<u128, anyhow::Error>;
+        _id: &Self::Identifier,
+        _mode: AggregatorReadMode,
+    ) -> anyhow::Result<u128> {
+        unimplemented!("Aggregator V2 is not yet supported")
+    }
 
     /// Returns a unique per-block identifier that can be used when creating a
     /// new aggregator.
-    fn generate_aggregator_id(&self) -> AggregatorID;
+    fn generate_aggregator_v2_id(&self) -> Self::Identifier {
+        unimplemented!("ID generation for Aggregator V2 is not yet supported")
+    }
+
+    /// Consumes a single delta and tries to materialize it with a given state
+    /// key. If materialization succeeds, a write op is produced. Otherwise, an
+    /// error VM status is returned.
+    // TODO(aggregator): This can be removed from the trait when `DeltaOp` is
```

Comment:
Why?

Reply:
Do you think that having it there is ok? Previously it has been a method on deltas, I am fine with both. Moved here mostly to avoid cyclic dependencies between aptos-aggregator and aptos-vm-types.